File: apps/web/src/types/gmail.ts
export interface GmailStatus {
  connected: boolean;
  enabled: boolean;
  connected_email: string | null;
  last_history_id: string | null;
  last_sync_at: string | null;
  mode: "real" | "mock";
}

export interface GmailSyncResult {
  processed: number;
  created_tickets: number;
  appended_to_threads: number;
  skipped_duplicates: number;
  mode: "real" | "mock";
}

export interface GmailTestResult {
  ok: boolean;
  connected_email: string | null;
  detail: string;
}

File: apps/api/src/star_itsm_api/schemas/gmail.py
from datetime import datetime

from pydantic import BaseModel, Field


class GmailOAuthStartResponse(BaseModel):
    authorize_url: str


class GmailOAuthCallbackResponse(BaseModel):
    connected: bool
    connected_email: str


class GmailStatusRead(BaseModel):
    connected: bool
    enabled: bool
    connected_email: str | None = None
    last_history_id: str | None = None
    last_sync_at: datetime | None = None
    mode: str = "real"


class GmailSettingsUpdate(BaseModel):
    enabled: bool | None = None


class GmailSyncResponse(BaseModel):
    processed: int
    created_tickets: int
    appended_to_threads: int
    skipped_duplicates: int
    mode: str = "real"


class GmailTestResponse(BaseModel):
    ok: bool
    connected_email: str | None = None
    detail: str = Field(default="OK")
